graphene_django_extras/utils.py

```diff
@@ -324,9 +318,24 @@ def recursive_params(
             [prefetch_related.append(x) for x in b if x not in prefetch_related]
             continue
 
+        field_def = get_field_def(root_info.schema, current_parent_type, field.name.value)
+        field_type = get_type(field_def.type)
+
+        optimization_hints = getattr(field_def.resolver, "optimization_hints", None)
+        if optimization_hints:
+            [
```

Comment:
Using a list comprehension only for its side effects is considered bad style. I see that it's doing this a few lines up as well, but that should be fixed.

If you work with `set` objects, you can skip the explicit loop entirely and express the intent of merging in the new entries directly:

```python
select_related = sorted(set(select_related) | optimization_hints.select_related)
```

The `sorted()` call is only necessary because the rest of the library doesn't pass those fields around as `set`s like it should.